### PR TITLE
Add send_mail to ediscovery

### PR DESCRIPTION
**Pull request by dashapaster**
add send_mail.sl to ediscovery for sending result of flow over mail
add retrive_result.sl for retrieving result and generating text for mail
add add_text_to_file.sl - for saving data in one file

### CODE REVIEW

File: content/io/cloudslang/haven_on_demand/ediscovery/ediscovery.sl [hunk 1]
```diff
@@ -86,16 +97,35 @@ flow:
...
-            publish:
-              - error_message: ${'step PRINT RESULTS was failed '+ str(error_message) if error_message!=None else ""}
+          publish:
+              - text_for_mail
+              - error_message: ${'step PRINT RESULTS was failed ' + reference + str(error_message) if error_message!=None else ""}
+
+      - read_from_file:
+          do:
+           file.read_from_file:
+               - file_path: "C:/Temp/result.txt"
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_on_demand/ediscovery/ediscovery.sl]: @dashapaster I'm not agree with hard-coded values... Please supply an input for it.

File: content/io/cloudslang/haven_on_demand/ediscovery/get_docs_references.sl [hunk 2]
```diff
@@ -32,6 +32,9 @@ operation:
           decoded = json.loads(json_input)
           list = decoded[key]
           references = []
+          if len(list) == 0:
+               is_error = True
+               error_message="invalid vlue"
```
Comment (mihaitusa) [inline content/io/cloudslang/haven_on_demand/ediscovery/get_docs_references.sl]: @dashapaster typo here

**Comment by mihaitusa** [comment]
@dashapaster Please handle the comments... after that (if everything is tested/works) I think it can be merged.

File: content/io/cloudslang/haven_on_demand/ediscovery/get_docs_references.sl (hunk 2 above)
Comment (dashapaster) [inline content/io/cloudslang/haven_on_demand/ediscovery/get_docs_references.sl]: done

File: content/io/cloudslang/haven_on_demand/ediscovery/ediscovery.sl (hunk 1 above)
Comment (dashapaster) [inline content/io/cloudslang/haven_on_demand/ediscovery/ediscovery.sl]: so, i think. i create input for this file

**Pull request merged**